build/teamcity-testrace.sh

```diff
@@ -24,7 +24,8 @@ else
 		echo "No changed packages; skipping race detector tests"
 		exit 0
 	fi
-	PKGSPEC=$(echo "${PKGSPEC}" | xargs -n1 dirname | sort | uniq | sed 's/^/.\//' | paste -sd " " -)
+	# Only keep the pathspecs for which there are go files remaining.
+	PKGSPEC=$(echo "${PKGSPEC}" | xargs -n1 dirname | sort | uniq | sed 's/^/.\//' | { while read path; do if ls "$path"/*.go >/dev/null 2>&1; then echo "$path"; fi; done; } | paste -sd " " -)
```

Comment:
does this work even when directories are fully removed? @danhhz ran into that recently

Reply:
Yes then the `ls` will fail and the thing is not listed.

Reply:
This PR is an example of it.